src/datasource.test.ts

```diff
@@ -61,61 +56,76 @@ describe('AthenaDatasource', () => {
       jsonData: { defaultRegion: 'testRegion', catalog: 'testCatalog', database: 'testDatabase' },
     } as unknown as DataSourceInstanceSettings<AthenaDataSourceOptions>;
     ctx.ds = new DataSource(ctx.instanceSettings);
-    runtime.setBackendSrv(backendSrv as runtime.BackendSrv);
+    ctx.ds.getResource = jest.fn().mockImplementation((path: string) => {
+      switch (path) {
+        case 'regions':
+          return Promise.resolve(setupRegionsResponse());
+      }
+      return Promise.resolve([]);
+    });
+
+    ctx.ds.postResource = jest.fn().mockImplementation((path: string) => {
+      switch (path) {
+        case 'catalogs':
+          return Promise.resolve(setupCatalogsResponse());
+        case 'databases':
+          return Promise.resolve(setupDatabasesResponse());
+        case 'tables':
+          return Promise.resolve(setupTablesResponse());
+        case 'columns':
+          return Promise.resolve(setupColumnsResponse());
+      }
+      return Promise.resolve([]);
+    });
   });
 
   describe('When performing getRegions', () => {
     it('should return a list of regions', async () => {
       const response = setupRegionsResponse();
-      backendSrv.get = jest.fn(() => Promise.resolve(response));
       const regionsResponse = await ctx.ds.getRegions();
 
       expect(regionsResponse).toHaveLength(response.length);
-      expect(regionsResponse).toBe(response);
+      expect(regionsResponse).toEqual(response);
```

Comment:
curious if this is the result of just a package update or if this is like meaningfully different in some way?

Reply:
the mock's response was previously set with `backendSrv.get = jest.fn(() => Promise.resolve(response));` (removed line 70), but now it's set up in the beforeEach so now it's not testing for reference equality anymore.